src/hammer-vlsi/par/openroad/__init__.py

```diff
@@ -80,152 +193,1177 @@ def export_config_outputs(self) -> Dict[str, Any]:
+                    layer = self.get_stackup().get_metal(pin_layer_name)
+                    if layer.direction==RoutingDirection.Horizontal:
+                        hor_layers.append(pin_layer_name)
+                    if layer.direction==RoutingDirection.Vertical:
+                        ver_layers.append(pin_layer_name)
+        
+        # both hor_layers and ver_layers arguments are required
+        # if missing, auto-choose one or both
+        if not (hor_layers and ver_layers):
+            self.logger.warning("Both horizontal and vertical pin layers should be specified. Hammer will auto-specify one or both.")
+            # choose first pin layer to be middle of stackup
+            #   or use pin layer in either hor_layers or ver_layers
+            pin_layer_names=["",""]
+            pin_layer_names[0]=all_metal_layer_names[int(len(all_metal_layer_names)/2)]
+            # pin_layer_name1=self.get_stackup().get_metal_by_index(int(len(self.get_stackup().metals)/2))
+            if (hor_layers): pin_layer_names[0]=hor_layers[0]
+            if (ver_layers): pin_layer_names[0]=ver_layers[0]
+            pin_layer_idx_1=all_metal_layer_names.index(pin_layer_names[0])
+            if (pin_layer_idx_1 < len(all_metal_layer_names)-1): 
+                pin_layer_names[1]=all_metal_layer_names[pin_layer_idx_1+1]
+            elif (pin_layer_idx_1 > 0): 
+                pin_layer_names[1]=all_metal_layer_names[pin_layer_idx_1-1]
+            else: # edge-case
+                pin_layer_names[1]=all_metal_layer_names[pin_layer_idx_1]
+            for pin_layer_name in pin_layer_names:
+                layer = self.get_stackup().get_metal(pin_layer_name)
+                if (layer.direction==RoutingDirection.Horizontal) and (pin_layer_name not in hor_layers):
+                    hor_layers.append(pin_layer_name)
+                if (layer.direction==RoutingDirection.Vertical)   and (pin_layer_name not in ver_layers):
+                    ver_layers.append(pin_layer_name)
+        return f"place_pins {random_arg} -hor_layers {{{' '.join(hor_layers)}}} -ver_layers {{{' '.join(ver_layers)}}} -exclude top:* -exclude right:* -exclude left:*"
+
     def place_pins(self) -> bool:
-        # TODO: currently using OpenROAD's default floorplan script
+        self.block_append(f"""
+        ################################################################
+        # IO Placement
+        {self.place_pins_tcl()}
+
+        write_def "{self.run_dir}/{self.top_module}_global_place.def"
+        """)
+        return True
+
+    def set_rc(self) -> bool:
+        self.block_append(f"""
+        ################################################################
+        # Repair max slew/cap/fanout violations and normalize slews
+        source "/tools/B/nayiri/openroad/hammer-openroad/src/hammer-vlsi/technology/sky130/extra/sky130hd.rc"
+        set_wire_rc -signal -layer "met2"
+        set_wire_rc -clock  -layer "met5"
```

Comment:
This particular method should come from the Sky130 plugin as written. Maybe solve this dependency in a future PR.